beacon-chain/blockchain/fork_choice.go

```diff
@@ -316,10 +319,43 @@ func (c *ChainService) attestationTargets(ctx context.Context, state *pb.BeaconS
 //        )
 func VoteCount(block *pb.BeaconBlock, state *pb.BeaconState, targets map[uint64]*pb.BeaconBlock, beaconDB *db.BeaconDB) (int, error) {
 	balances := 0
+	var ancestor *pb.BeaconBlock
+	var err error
+
 	for validatorIndex, targetBlock := range targets {
-		ancestor, err := BlockAncestor(targetBlock, block.Slot, beaconDB)
-		if err != nil {
-			return 0, err
+		if featureconfig.FeatureConfig().EnableBlockAncestorCache {
+			// if block ancestor cache was enabled, use target block hash + ancestor block height as cache key.
+			targetHash, err := hashutil.HashBeaconBlock(targetBlock)
```

Comment:
Can this block be abstracted into its own small function? It makes the vote count func quite big and hard to follow

Reply:
will do!